### DISCUSSION TITLE
Initialization of spray droplets

### DISCUSSION THREAD

**Discussion by XUFEI-Sai** [AMReX-Combustion/PelePhysics#537] [Q&A] [answered]
Hello, I am interested in spray combustion recently. I read some cases of spray in PeleC. It is known that the liquid fuel can be injected into the computational domain through injection in `SprayParticlesInitInsert.cpp`. Now I want the flow field to evenly distribute some droplets during initialization. What functions and variables should be defined?

**Comment by baperry2** [comment, marked as answer]
See `PeleC/Exec/RegTests/Spray-Conv` for an example where a uniform distribution of spray droplets is initialized.

**Reply by XUFEI-Sai** [reply]
Thank you very much. I saw that `uniformSprayInit` can be used in `InitiSprayParticles`. But if I want the initial spray droplets to only exist in a certain area of the calculation domain, can I set this?

**Reply by baperry2** [reply]
There isn't a built in way to do that. The `InitSprayParticles` function gives flexibility to define any initial particle distribution you want, but it must be done manually. In this case you could probably fairly easily adapt the `uniformSprayInit` function to put particles only in a subregion of the domain rather than the whole domain.

**Reply by XUFEI-Sai** [reply]
Okay, I'll have a try. Thank you again for your help.

**Comment by RSuryaNarayan** [comment]
Hello, I have a follow-up question here: is it possible to seed particles at an inlet of the domain uniformly as a boundary condition? Or is it only possible presently to inject via the SprayJet class? Thanks a lot in advance!

**Reply by baperry2** [reply]
I believe you could set up a uniform injection spray boundary using the SprayJet class by having it cover the whole domain. 

@SreejithNREL - would anything you've done for Spray injection for SAF be useful for this>

**Reply by RSuryaNarayan** [reply]
that makes sense, I'll try it out with the spray-jet case. I guess to be safe I'd have to use ```theta_J``` here to be ```pi``` to enable a flat injection at the inlet ?

![image](https://github.com/user-attachments/assets/8143e398-09f6-4540-b65d-a855963ba1bb)

**Reply by RSuryaNarayan** [reply]
I remember trying this out once before though but the angles of the spray still remained..will try again

**Reply by SreejithNREL** [reply]
I think it should be doable with the current SprayJet class itself.

**Reply by RSuryaNarayan** [reply]
@SreejithNREL @baperry2 thanks. this trick works just fine! I used a ```jet.spread_angle=90``` so that I get a flat jet.
